packages/venia-concept/src/util/makeUrl.js

```diff
@@ -1,64 +1,56 @@
-import memoize from 'lodash/memoize';
-
-// Tests if a URL begins with `http://` or `https://` or `data:`
-const absoluteUrl = /^(data|https|http)?:/i;
-
-// Simple path joiner that guarantees one and only one slash between segments
+// ensure there's exactly one slash between segments
 const joinUrls = (base, url) =>
     (base.endsWith('/') ? base.slice(0, -1) : base) +
     '/' +
     (url.startsWith('/') ? url.slice(1) : url);
 
-// TODO: make this even more dynamic, like an open registry
-const mediaPathPrefixes = {
-    'image-product':
+const mediaBases = new Map()
+    .set(
+        'image-product',
         process.env.MAGENTO_BACKEND_MEDIA_PATH_PRODUCT ||
-        '/media/catalog/product',
-    'image-category':
-        process.env.MAGENTO_BACKEND_MEDIA_PATH_CATALOG ||
-        '/media/catalog/category'
-};
+            '/media/catalog/product'
+    )
+    .set(
+        'image-category',
+        process.env.MAGENTO_BACKEND_MEDIA_PATH_CATEGORY ||
+            '/media/catalog/category'
+    );
+
+const resizeBase = joinUrls(
+    process.env.IMAGE_SERVICE_PATH || '/img/',
+    '/resize/'
+);
+
+const makeOptimizedUrl = (path, { type, width } = {}) => {
```

Comment:
A jsdoc style comment explaining the functionality would be really helpful here!